### PR TITLE
Add tests, output visualization

### PR DESCRIPTION
**Pull request by mciantyre**
Added some tests during my first review. Also, added a quick binary so that I could visualize the generated linker script. Run

```
cargo run --bin render
```

to see what `LinkerScript` produces.

### CODE REVIEW

File: src/bin/render.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    let flash = ls.region(FLASH, 0x0, 512).unwrap();
+    let ram = ls.region(RAM, 0x20000000, 128).unwrap();
+    ls.stack(ram.clone()).unwrap();
+    ls.heap(ram.clone()).unwrap();
+    ls.boot_config(512, "fcb", flash.clone()).unwrap();
+    ls.vector_table(ram.clone(), Some(flash.clone())).unwrap();
+    ls.text(flash.clone(), None).unwrap();
+    ls.data(false, ram.clone(), Some(flash.clone())).unwrap();
+    ls.rodata(false, flash.clone(), None).unwrap();
+    ls.bss(false, ram.clone(), Some(flash.clone())).unwrap();
```
Comment (mciantyre) [inline src/bin/render.rs:20]: Any reason to favor the `.clone()` calls over immutable borrows?

```rust
ls.bss(false, &ram, Some(&flash)).unwrap();
```

is fewer characters to type.

File: src/lib.rs [hunk 2]
```diff
@@ -362,21 +362,28 @@ impl<W: Word> LinkerScript<W> {
...
+    /// the current working directory.
     pub fn generate(self) -> Result<()> {
-        const REQ_SEC_NAMES: [&str; 5] = ["stack", "vector_table", "text", "data", "bss"];
+        let mut link_x = File::create("link.x")?;
+        self.write(&mut link_x)
+    }
+
+    /// Write the linker script into the writer, `link_x`
+    pub fn write<Wr: Write>(self, link_x: &mut Wr) -> Result<()> {
+        const REQ_SEC_NAMES: [&str; 6] = ["stack", "vector_table", "text", "data", "rodata", "bss"];
```
Comment (mciantyre) [inline src/lib.rs:375]: `rodata` is documented as a required section. I added it to this list, and checked it with some tests (below).

**Comment by mciantyre** [comment]
Here's part of the output from the renderer. Does the `.heap` section overwrite the `.stack` section? Should these sections be generated with the [updates to the `__*_used` symbols](https://github.com/imxrt-rs/imxrt-rt-gen/blob/6d43ebebf7347b5d82ae1a8b98dca3714938dec7/src/generate/link.rs#L35)?

```
	.stack :
	{
		. = __RAM_origin + __RAM_used;
		. = ALIGN(4);
		__end_stack = .;
		. = __RAM_origin + __RAM_size;
		__start_stack = .;
	} > RAM

	.heap :
	{
		. = __RAM_origin + __RAM_used;
		. = ALIGN(4);
		__start_heap = .;
		. = __RAM_origin + __RAM_size;
		__end_heap = .;
	} > RAM
```

**Comment by teburd** [comment]
The heap and stack collide somewhere in the middle, the heap starts at the first free ram address going up, the stack starts at the last ram address going down.

Jorge has a nifty double linking techinque that avoids this potential data overwriting issue in a very nice way

https://blog.japaric.io/stack-overflow-protection/

File: src/bin/render.rs (hunk 1 above)
Comment (teburd) [reply src/bin/render.rs:20]: Immutable borrows are fine, I had a copyable integer id being passed around before switching to a string identifier, one of those times where I wish I had a copyable string type, these strings aren't long.

**Pull request merged**